### PR TITLE
Rename NU4 to Canopy in constant and function names

### PR DESCRIPTION
**Pull request by daira**
Signed-off-by: Daira Hopwood <daira@jacaranda.org>

### CODE REVIEW

File: qa/rpc-tests/upgrade_golden.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ class UpgradeGoldenTest(BitcoinTestFramework):
     def setup_chain(self):
         self.upgrades = [ Upgrade(35, os.path.dirname(os.path.realpath(__file__))+"/golden/blossom.tar.gz", HAS_BLOSSOM)
                         , Upgrade(45, os.path.dirname(os.path.realpath(__file__))+"/golden/heartwood.tar.gz", HAS_HEARTWOOD)
-                        , Upgrade(55, os.path.dirname(os.path.realpath(__file__))+"/golden/nu4.tar.gz", HAS_NU4)
+                        , Upgrade(55, os.path.dirname(os.path.realpath(__file__))+"/golden/nu4.tar.gz", HAS_CANOPY)
```
Comment (daira) [inline qa/rpc-tests/upgrade_golden.py:36]: I was unsure whether to also update the tarfile name here.

File: src/consensus/upgrades.cpp [hunk 2]
```diff
@@ -41,8 +41,8 @@ const struct NUInfo NetworkUpgradeInfo[Consensus::MAX_NETWORK_UPGRADES] = {
     },
     {
         /*.nBranchId =*/ 0xe9ff75a6,
-        /*.strName =*/ "NU4",
-        /*.strInfo =*/ "See https://z.cash/upgrade/nu4/ for details.",
+        /*.strName =*/ "Canopy",
+        /*.strInfo =*/ "See https://z.cash/upgrade/canopy/ for details.",
```
Comment (daira) [inline src/consensus/upgrades.cpp:45]: This URL isn't live yet (but neither was https://z.cash/upgrade/nu4/ ).

File: qa/rpc-tests/upgrade_golden.py (hunk 1 above)
Comment (nuttycom) [reply qa/rpc-tests/upgrade_golden.py:36]: Yes, since we're not yet persisting the tarfile in git, now is the time to rename it.

**Comment by ebfull** [comment]
ACK

@zkbot r+

**Comment by str4d** [comment]
@zkbot r+

**Comment by str4d** [comment]
@zkbot r+

**Pull request merged**